features/ubuntu_pro.feature

```diff
@@ -174,22 +174,84 @@ Feature: Command behaviour when auto-attached in an ubuntu PRO image
 
     Examples: ubuntu release
       | release | machine_type |
-      | xenial  | aws.pro      |
       | xenial  | azure.pro    |
       | xenial  | gcp.pro      |
-      | bionic  | aws.pro      |
       | bionic  | azure.pro    |
       | bionic  | gcp.pro      |
-      | focal   | aws.pro      |
       | focal   | azure.pro    |
       | focal   | gcp.pro      |
-      | jammy   | aws.pro      |
       | jammy   | azure.pro    |
       | jammy   | gcp.pro      |
-      | noble   | aws.pro      |
       | noble   | azure.pro    |
       | noble   | gcp.pro      |
 
+  Scenario Outline: Auto-attach service works on Pro Machine on aws.pro
```

Comment:
so this test ensures backwards compatability with existing pro images? and `Scenario Outline: Auto-attach service works on Pro Machine on aws.generic` tests the new functionality right?

Reply:
`Scenario Outline: Auto-attach service works on Pro Machine on aws.pro` and
`Scenario Outline: Auto-attach service works on Pro Machine on aws.generic` test that pro images and generic images behave correctly with `ubuntu-advantage-pro` installed. 

This service is pre-installed in every Ubuntu Pro image. And at some point in the future, likely, AWS generic images will start shipping `ubuntu-advantage-pro` and not auto-attaching, thanks to this PR.

Reply:
For more context see the attached jira and spec.